homeassistant/components/unifi/device_tracker.py

```diff
@@ -182,18 +199,87 @@ def available(self) -> bool:
 
     @property
     def device_info(self):
-        """Return a device description for device registry."""
+        """Return a client description for device registry."""
         return {
             'connections': {(CONNECTION_NETWORK_MAC, self.client.mac)}
         }
 
     @property
     def device_state_attributes(self):
-        """Return the device state attributes."""
+        """Return the client state attributes."""
         attributes = {}
 
         for variable in DEVICE_ATTRIBUTES:
             if variable in self.client.raw:
                 attributes[variable] = self.client.raw[variable]
 
         return attributes
+
+
+class UniFiDeviceTracker(ScannerEntity):
+    """Representation of a network infrastructure device."""
+
+    def __init__(self, device, controller):
+        """Set up tracked device."""
+        self.device = device
+        self.controller = controller
+
+    async def async_update(self):
+        """Synchronize state with controller."""
+        await self.controller.request_update()
+
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        detection_time = self.controller.unifi_config.get(
+            CONF_DETECTION_TIME, DEFAULT_DETECTION_TIME)
+
+        if (dt_util.utcnow() - dt_util.utc_from_timestamp(float(
+                self.device.last_seen))) < detection_time:
+            return True
+        return False
+
+    @property
+    def source_type(self):
+        """Return the source type of the device."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self.device.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this device."""
+        return self.device.mac
+
+    @property
+    def available(self) -> bool:
+        """Return if controller is available."""
+        return self.controller.available
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            'connections': {(CONNECTION_NETWORK_MAC, self.device.mac)},
+            'manufacturer': ATTR_MANUFACTURER,
+            'model': self.device.model,
+            'name': self.device.name,
+            'sw_version': self.device.version
+        }
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {}
+
+        attributes['board_rev'] = self.device.board_rev
```

Comment:
Is this needed for automations?

Reply:
No, but it is information not possible to expose through device registry

Reply:
We shouldn't add static information here. If we want to expose more info as device info we should bring it up as an architecture issue.

Reply:
Allright, removing